src/client.ts

```diff
@@ -335,6 +341,152 @@ export class McpdClient {
     return allTools;
   }
 
+  /**
+   * Get prompts from all (or specific) MCP servers with namespaced names.
+   *
+   * IMPORTANT: Prompt names are transformed to `serverName__promptName` format to:
+   * 1. Prevent naming clashes when aggregating prompts from multiple servers
+   * 2. Identify which server each prompt belongs to
+   *
+   * This method automatically filters out unhealthy servers by checking their health
+   * status before fetching prompts. Unhealthy servers are silently skipped to ensure
+   * the method returns quickly without waiting for timeouts on failed servers.
+   *
+   * Servers that don't implement prompts (return 501 Not Implemented) are silently
+   * skipped, allowing this method to work with mixed server types.
+   *
+   * Prompt fetches from multiple servers are executed concurrently for optimal performance.
+   *
+   * @param options - Optional configuration
+   * @param options.servers - Array of server names to include. If not specified, includes all servers.
+   * @returns Array of prompt schemas with transformed names (serverName__promptName). Only includes prompts from healthy servers that support them.
+   * @throws {ConnectionError} If unable to connect to the mcpd daemon
+   * @throws {TimeoutError} If requests to the daemon time out
+   * @throws {AuthenticationError} If API key authentication fails
+   * @throws {McpdError} If health check or initial server listing fails
+   *
+   * @example
+   * ```typescript
+   * // Get all prompts from all servers
+   * const allPrompts = await client.getPrompts();
+   * // Returns: [
+   * //   { name: "github__create_pr", description: "...", arguments: [...] },
+   * //   { name: "notion__create_page", description: "...", arguments: [...] }
+   * // ]
+   *
+   * // Get prompts from specific servers only
+   * const somePrompts = await client.getPrompts({ servers: ['github', 'notion'] });
+   *
+   * // Original prompt name "create_pr" becomes "github__create_pr"
+   * // This prevents clashes if multiple servers have prompts with the same name
+   * ```
+   */
+  async getPrompts(options?: { servers?: string[] }): Promise<Prompt[]> {
+    const { servers } = options || {};
+
+    // Determine which servers to query.
+    const serverNames =
+      servers && servers.length > 0 ? servers : await this.listServers();
+
+    // Get health status for all servers.
+    const healthMap = await this.getServerHealth();
+
+    // Filter to only healthy servers.
+    const healthyServers = serverNames.filter((name) => {
+      const health = healthMap[name];
+      return health && HealthStatusHelpers.isHealthy(health.status);
+    });
```

Comment:
nit: considering this is done in other places in the project, it can be extracted to a helper function called `getHealthyServers` or so to reduce the noise in the function and make its purpose clear when reading it